File: prolog/numbers/test/numbers_recursive_test.pl
:- use_module(library(plunit)).
:- consult('../src/numbers_recursive.pl').

:- begin_tests(recursive_tests).
test(fibonacci) :-
    fibonacci(5, Result),
    assertion(Result == 5).

test(factorial) :-
    factorial(5, Result),
    assertion(Result == 120).

test(sum_numbers) :-
    sum_numbers(5, Sum),
    assertion(Sum == 15).

:- end_tests(recursive_tests).

:- initialization(run_tests(recursive_tests), main).

File: f#/numbers/test/NumbersRecursiveTest.fs
namespace test

open NUnit.Framework
open src.NumbersRecursive

[<TestFixture>]
module RecursiveTests =

    [<Test>]
    let ``Test Fibonacci recursive`` () =
        Assert.AreEqual(5, fibonacci 5)

    [<Test>]
    let ``Test Factorial recursive`` () =
        Assert.AreEqual(120, factorial 5)

    [<Test>]
    let ``Test sum_numbers recursive`` () =
        Assert.AreEqual(15, sum_numbers 5)